src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java

```diff
@@ -104,6 +119,29 @@ public void setNamespace(String namespace) {
     this.namespaces = StringUtils.isBlank(namespace)?null:namespace.split(" ");
   }
 
+  // https://wiki.jenkins-ci.org/display/JENKINS/Credentials+Plugin
+  // http://javadoc.jenkins-ci.org/credentials/com/cloudbees/plugins/credentials/common/AbstractIdCredentialsListBoxModel.html
+  // https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java
+  public static ListBoxModel doFillCredentialsIdItems(String credentialsId) {
+    if (credentialsId == null) {
+      credentialsId = "";
+    }
+
+    if (!Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER)) {
+      // Important! Otherwise you expose credentials metadata to random web requests.
+      return new StandardListBoxModel().includeCurrentValue(credentialsId);
+    }
+
+    return new StandardListBoxModel()
+            .includeEmptyValue()
+            .includeAs(ACL.SYSTEM, Jenkins.getInstance(), OpenShiftTokenCredentials.class)
+            //.includeAs(ACL.SYSTEM, Jenkins.getInstance(), StandardUsernamePasswordCredentials.class)
+            // .includeAs(ACL.SYSTEM, Jenkins.getInstance(), StandardCertificateCredentials.class)
+            // TODO: Make own type for token or use the existing token generator auth type used by sync plugin? or kubernetes?
```

Comment:
The sharing of credential related artifacts also arose when @jupierce first put together the client plugin as well.  Unless @bparees disagrees, go ahead and get a card opened to look at component refactoring and sharing across all of our plugins as a MVP stmt, with extra credit for factoring in the k8s cloud plugin (first as a dependency, second if we want to suggest changes to it upstream).  Perhaps reference the card in place of this TODO.